### DISCUSSION TITLE
How to use NXP HAL Directly

### DISCUSSION THREAD

**Discussion by PxGreen101** [zephyrproject-rtos/zephyr#43671] [Q&A] [answered]
Is there a way to use the NXP HAL directly? For example I would like to init the SCTimer module using the init function inside fsl_ctimer.h 
/* Initialize SCTimer module */
	SCTIMER_Init(SCT0, &sctimerInfo);

For STM32 this was possible using the information provided in \modules\hal\stm32\README.rst where I could use a hardware timer directly using the USE_STM32_LL_TIM

I would like to find out if something like this is possible for NXP.

**Comment by PxGreen101** [comment, marked as answer]
NVM - figured it out.

**Reply by mirhamza708** [reply]
can you please share how you did it?

**Reply by DerekSnell** [reply]
Hi @mirhamza708 ,
Please see [Using MCUXpresso SDK drivers in Zephyr app](https://community.nxp.com/t5/Zephyr-Project-Knowledge-Base/Using-MCUXpresso-SDK-drivers-in-a-Zephyr-application/ta-p/2077778).  Best regards
